docs/architecture/design.md

```diff
@@ -0,0 +1,28 @@
+# Architecture
+
+SC4SNMP is deployed using a Kubernetes distribution, typically MicroK8s,
+that's designed to be a low-touch experience for integration with sensitive
+edge network devices. It will typically be deployed in the same network
+management zone as the monitored devices and separated from Splunk by an
+existing firewall.
+
+![image](../images/sc4snmp_deployment.png)
+
+
+## High-level Design 
+
+SC4SNMP has two main purposes. The first one is used to collect SNMP data from network 
+devices according to planned schedules and the second one is responsible for listening to SNMP traps.
+
+![image](../images/sc4snmp_architecture.png)
+
+Diagram above present high level architecture of Splunk Connector for SNMP, it contains following components:
+
+- **UI** - user interface for configuring the SC4SNMP profiles, groups, and inventory.
+- **Poller** - responsible for getting selected data from SNMP agents in set periods of time.
```

Comment:
It's good that we mentioned about that periodic tasks. Probably have a sense to add few details that we using Celery & Celery beat for that

Reply:
that's on the diagram, I can add the section of describing what's inside every component but I wasn't sure if that isn't too much details that won't matter for customers